source/common/stats/symbol_table_impl.h

```diff
@@ -630,5 +630,44 @@ class StatNameStorageSet {
   HashSet hash_set_;
 };
 
+// Captures StatNames for lookup by string, keeping two maps: a map of
+// 'built-ins' that is expected to be populated during initialization, and a map
+// of dynamically discovered names. The latter map is protected by a mutex, and
+// can be mutated at runtime.
+class StatNameSet {
+public:
+  explicit StatNameSet(SymbolTable& symbol_table);
+
+  /**
+   * Adds a string to the builtin map, which is not mutex protected. This
+   * map is always consulted first as a hit there means no lock is required.
+   */
+  void rememberBuiltin(absl::string_view str);
```

Comment:
Maybe reiterate the bit about "should be called during initialization" for the comment here? Might also be nice to have a note what exactly initialization means: is it just before any stats are accessed?